builtin/credential/ldap/backend.go

```diff
@@ -132,6 +132,15 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		return nil, logical.ErrorResponse(err.Error()), nil, nil
 	}
 
+	if cfg.AnonymousGroupSearch {
```

Comment:
Would you mind to explain this part a bit further? The below code is exactly the same we already execute on line 80: https://github.com/hashicorp/vault/blob/4fbf1f41dc8d01da4010c974d823fe9d3be4e0d1/builtin/credential/ldap/backend.go#L80-L86

Reply:
With google LDAP there's no way to query for groups a user is logged in as while logged in as that user. But there's no way to verify a user's login credentials without binding as that user. At least no way I could find to do it.

Reason I added this was so that you could do the first bind as the logged in user, verify that their credentials are correct - then drop to the anonymous bind with client certificates just for the group membership search.

Reply:
Ah, gotcha! That sounds very google LDAP specific to me? I wonder if we should rename `anonymous_group_search` to `google_anonymous_group_search` or something similar? `anonymous_group_search` sounds to me like the credentials are never used at all.

Reply:
Sure, that makes sense and I agree. I will make that change :+1:

Reply:
The problem is if this behavior is ever necessary with an ldap server that isn't Google, now the option name is confusing/limiting.

Reply:
Yes, @michelvocks made a convincing argument for the name change. I will change it.

Reply:
Yes, and I'm suggesting that it should not be name that was discussed.

Reply:
Oh my bad, I misunderstood your point then. I'm happy either way, right now we only know that Google LDAP services need this option - but potentially I guess it is possible that others would need it.